opentelemetry-auto-instrumentation/lib/opentelemetry-auto-instrumentation.rb

```diff
@@ -0,0 +1,149 @@
+    'action_mailer' => 'OpenTelemetry::Instrumentation::ActionMailer',
+    'aws_sdk' => 'OpenTelemetry::Instrumentation::AwsSdk',
+    'aws_lambda' => 'OpenTelemetry::Instrumentation::AwsLambda',
+    'bunny' => 'OpenTelemetry::Instrumentation::Bunny',
+    'lmdb' => 'OpenTelemetry::Instrumentation::LMDB',
+    'http' => 'OpenTelemetry::Instrumentation::HTTP',
+    'koala' => 'OpenTelemetry::Instrumentation::Koala',
+    'active_model_serializers' => 'OpenTelemetry::Instrumentation::ActiveModelSerializers',
+    'concurrent_ruby' => 'OpenTelemetry::Instrumentation::ConcurrentRuby',
+    'dalli' => 'OpenTelemetry::Instrumentation::Dalli',
+    'delayed_job' => 'OpenTelemetry::Instrumentation::DelayedJob',
+    'ethon' => 'OpenTelemetry::Instrumentation::Ethon',
+    'excon' => 'OpenTelemetry::Instrumentation::Excon',
+    'faraday' => 'OpenTelemetry::Instrumentation::Faraday',
+    'grape' => 'OpenTelemetry::Instrumentation::Grape',
+    'graphql' => 'OpenTelemetry::Instrumentation::GraphQL',
+    'http_client' => 'OpenTelemetry::Instrumentation::HttpClient',
+    'mongo' => 'OpenTelemetry::Instrumentation::Mongo',
+    'mysql2' => 'OpenTelemetry::Instrumentation::Mysql2',
+    'net_http' => 'OpenTelemetry::Instrumentation::Net::HTTP',
+    'pg' => 'OpenTelemetry::Instrumentation::PG',
+    'que' => 'OpenTelemetry::Instrumentation::Que',
+    'racecar' => 'OpenTelemetry::Instrumentation::Racecar',
+    'rack' => 'OpenTelemetry::Instrumentation::Rack',
+    'rails' => 'OpenTelemetry::Instrumentation::Rails',
+    'rake' => 'OpenTelemetry::Instrumentation::Rake',
+    'rdkafka' => 'OpenTelemetry::Instrumentation::Rdkafka',
+    'redis' => 'OpenTelemetry::Instrumentation::Redis',
+    'restclient' => 'OpenTelemetry::Instrumentation::RestClient',
+    'resque' => 'OpenTelemetry::Instrumentation::Resque',
+    'ruby_kafka' => 'OpenTelemetry::Instrumentation::RubyKafka',
+    'sidekiq' => 'OpenTelemetry::Instrumentation::Sidekiq',
+    'sinatra' => 'OpenTelemetry::Instrumentation::Sinatra'
+  }.freeze
+
+  def detect_resource_from_env
+    env = ENV['OTEL_RUBY_RESOURCE_DETECTORS'].to_s
+    additional_resource = ::OpenTelemetry::SDK::Resources::Resource.create({})
+
+    env.split(',').each do |detector|
+      case detector
+      when 'container'
+        additional_resource = additional_resource.merge(::OpenTelemetry::Resource::Detector::Container.detect) if defined? ::OpenTelemetry::Resource::Detector::Container
+      when 'google_cloud_platform'
+        additional_resource = additional_resource.merge(::OpenTelemetry::Resource::Detector::GoogleCloudPlatform.detect) if defined? ::OpenTelemetry::Resource::Detector::GoogleCloudPlatform
+      when 'azure'
+        additional_resource = additional_resource.merge(::OpenTelemetry::Resource::Detector::Azure.detect) if defined? ::OpenTelemetry::Resource::Detector::Azure
+      end
+    end
+
+    additional_resource
+  end
+
+  def determine_enabled_instrumentation
```

Comment:
I am a little nervous about prepending all of these methods and adding them bundler. 

We sholud follow the Open Closed principle here and move these into a separate module instead and delegate to them instead of directly mixing them in. 

In other words the OTelBundlerPatch should only update the `require` method. The `require_otel` methods et.al. should be moved to a nested module like `OTelBundlerPath::Initializer.require_otel` or something like that. 

 That will keep the `Bundler` object from having unexpected methods on it or possilbe method collision names.